### PR TITLE
K epsilon (draft) (wip)

### PR DESCRIPTION
**Pull request by tanoret**
This is a work in progress. Still, tests and docs are missing.

### CODE REVIEW

File: modules/navier_stokes/src/fvbcs/NSFVTKEDWallFunctionReichardtBC.C [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    const ADReal b_c = 1/karman_cte * (std::log(E*dist/_mu(makeElemArg(&_current_elem))) + 1.0);
+    const ADReal c_c = parallel_speed;
+    u_star = (-b_c + std::sqrt(std::pow(b_c,2)+4.0*a_c*c_c))/(2.0 * a_c);
+  }
+  else
+    u_star = findUStar(_mu(makeElemArg(&_current_elem)), _rho(makeElemArg(&_current_elem)), parallel_speed, dist);
+  
+  ADReal y_plus = dist * u_star * _rho(makeElemArg(&_current_elem)) / _mu(makeElemArg(&_current_elem));
+
+  // _console << "Mu: " << _mu(makeElemArg(&_current_elem)) << std::endl;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvbcs/NSFVTKEDWallFunctionReichardtBC.C:97]: Remove unused code

File: modules/navier_stokes/src/fvbcs/NSFVTKEDWallFunctionReichardtBC.C [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  Real dist = std::abs((fi.elemCentroid() - fi.faceCentroid()) * fi.normal());
+  const Elem & _current_elem = fi.elem();
+
+  // Assign boundary weights to element
+  Real weight = 0.0;
+  for (unsigned int i_side = 0; i_side < _current_elem.n_sides(); ++i_side)
+  {
+    weight += static_cast<Real>(_subproblem.mesh().getBoundaryIDs(&_current_elem, i_side).size());
+  }
+  //_console << "Weight: " << weight << std::endl;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvbcs/NSFVTKEDWallFunctionReichardtBC.C:70]: remove this

File: modules/navier_stokes/src/fvbcs/NSFVTKEDWallFunctionReichardtBC.C [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    _w_var(params.isParamValid("w")
+               ? &(getFunctor<ADReal>("w"))
+               : nullptr),
+    _rho(getFunctor<ADReal>(NS::density)),
+    _mu(getFunctor<ADReal>("mu")),
+    _mu_t(getFunctor<ADReal>("mu_t")),
+    _k(getFunctor<ADReal>("k")),
+    _C_mu(getFunctor<ADReal>("C_mu")),
+    _linearized_yplus(getParam<bool>("linearized_yplus")),
+    _min_mixing_length(getParam<Real>("min_mixing_length"))
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvbcs/NSFVTKEDWallFunctionReichardtBC.C:54]: Is this used for anything?

File: modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionBC.C [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+NSFVTKEWallFunctionBC::NSFVTKEWallFunctionBC(const InputParameters & parameters)
+  : FVFluxBC(parameters),
+    _eps(getFunctor<ADReal>(NS::porosity))
+{
+}
+
+ADReal
+NSFVTKEWallFunctionBC::computeQpResidual()
+{
+  // return - _eps[_qp] * 0.0;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionBC.C:34]: Remove unused code

File: modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionReichardtBC.C [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+             ? &(getFunctor<ADReal>("v"))
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? &(getFunctor<ADReal>("w"))
+               : nullptr),
+    _rho(getFunctor<ADReal>(NS::density)),
+    _mu(getFunctor<ADReal>("mu")),
+    _C_mu(getFunctor<ADReal>("C_mu")),
+    _linearized_yplus(getParam<bool>("linearized_yplus")),
+    _min_mixing_length(getParam<Real>("min_mixing_length"))
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionReichardtBC.C:50]: Is this used for anything?

File: modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionReichardtBC.C [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      velocity(2) = (*_w_var)(makeElemArg(&_current_elem));
+
+    // Compute the velocity and direction of the velocity component that is parallel to the wall
+    ADReal parallel_speed  = (velocity - velocity * (fi.normal()) * (fi.normal())).norm();
+
+    ADReal u_star;
+    if (_linearized_yplus)
+    {
+      constexpr Real karman_cte = 0.4187;
+      constexpr Real E = 9.793;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionReichardtBC.C:74]: Why constexpr, do we need this compile time? Also, do you think the user should be allowed to set this?

File: modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionReichardtBC.C [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      const ADReal c_c = parallel_speed;
+      u_star = (-b_c + std::sqrt(std::pow(b_c,2)+4.0*a_c*c_c))/(2.0 * a_c);
+    }
+    else
+      u_star = findUStar(_mu(makeElemArg(&_current_elem)), _rho(makeElemArg(&_current_elem)), parallel_speed, dist);
+
+    // Realizable constraint
+    ADReal ralizable_constraint = std::pow(u_star / dist, 2);
+
+    _console << "Realizable constraint: " << ralizable_constraint.value() << std::endl;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvbcs/NSFVTKEWallFunctionReichardtBC.C:86]: Remove console output here.

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("w", 0))
+               : nullptr),
+    _harmonic_interpolation(getParam<bool>("harmonic_interpolation"))
+{
+  if ((_var.faceInterpolationMethod() == Moose::FV::InterpMethod::SkewCorrectedAverage) &&
+      (_tid == 0))
+  {
+    auto & factory = _app.getFactory();
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:48]: This has changed lately, check `FVDiffusion` for an example. It will be something like this:

```
if ((_var.faceInterpolationMethod() == Moose::FV::InterpMethod::SkewCorrectedAverage) &&
      (_tid == 0))
    adjustRMGhostLayers(std::max((unsigned short)(3), _pars.get<unsigned short>("ghost_layers")));
```

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+    if (!_app.addRelationshipManager(rm_obj))
+      factory.releaseSharedObjects(*rm_obj);
+  }
+}
+
+ADReal
+INSFVMomentumDiffusionComplete::computeStrongResidual()
+{
+  /// ------------------ Useful variables ----------------------
+  // Booleans to determine if a face have an associated element and a neighbor
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:76]:
```suggestion
  // Booleans to determine if a face has an associated element and a neighbor
```

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+  ADRealTensorValue gradient;
+  if (_dim == 1)
+  {
+    //const auto & grad_u = _u_var->adGradSln(face_info, skewness_correction);
+    const auto & grad_u = _u_var->gradient(face);
+    gradient = ADRealTensorValue(grad_u);
+  }
+  else if (_dim == 2)
+  {
+    // const auto & grad_u = _u_var->adGradSln(face_info, skewness_correction);
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:110]: Remove unused code

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+
+  /// ---------------------------------------------------------
+
+  /// ------ Interpolating the diffusion coefficient with different methods ------
+
+  auto mu_elem = has_elem ? _mu(elem_face) : _mu(neighbor_face);
+  auto mu_neighbor = has_neighbor ? _mu(neighbor_face) : _mu(elem_face);
+
+  const Real distance_to_neighbor = _face_info->dCN().norm();
+  const Real distance_to_element   = _face_info->eCN().norm();
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:138]: eCN() is dCN/mag(dCN), so a unit vector.

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+  const Real total_distance        = distance_to_neighbor + distance_to_element;
+
+  // Ideally, we would do the following, but we optimize some operations:
+    //   const Real coef_elem = distance_to_neighbor / total_distance;
+    //   const Real coef_neighbor = distance_to_element / total_distance;
+
+    //   ADReal mu_face;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:145]: Remove unused code

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+    //   if (_harmonic_interpolation)
+    //     mu_face = 1.0 / (coef_neighbor / mu_neighbor + coef_elem / mu_elem);
+    //   else
+    //     mu_face = coef_neighbor * mu_neighbor + coef_elem * mu_elem;
+
+  const Real coef_elem = distance_to_neighbor / total_distance;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:151]: This is already in faceInfo `const Real coef_elem = _face_info->gC()`

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+
+  ADReal mu_face;
+  Real protection_mu = 1e-12;
+  if (_harmonic_interpolation)
+  {
+    const auto inv_mu_neighbor = 1.0 / (mu_neighbor + protection_mu);
+    mu_face = 1.0 / (coef_elem * (1/(mu_elem + protection_mu) - inv_mu_neighbor) + inv_mu_neighbor);
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:158]: Harmonic interpolation is now available in `MathFVUtils.h`

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+  }
+  else
+    mu_face = coef_elem * (mu_elem - mu_neighbor) + mu_neighbor;
+
+  // if (mu_elem < 1e10)
+  //   _console << "Mu Elem: " << mu_elem << std::endl;
+  // if (mu_neighbor < 1e10)
+  //   _console << "Mu Neigh: " << mu_neighbor << std::endl;
+  // if (mu_face < 1e10)
+  //   _console << "Mu Face: " << mu_face << std::endl;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:168]: Remove unused code

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+#include "INSFVRhieChowInterpolator.h"
+#include "NS.h"
+#include "SystemBase.h"
+#include "RelationshipManager.h"
+#include "Factory.h"
+
+registerMooseObject("NavierStokesApp", INSFVMomentumDiffusionComplete);
+
+InputParameters
+INSFVMomentumDiffusionComplete::validParams()
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:20]: shouldn't this be: 
`nabla(mu*(grad(u) + grad(phi)^T-2/3div(u)*I))`? I suppose the div(u) part will only show up if we do weakly compressible

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+
+  ADReal mu_face;
+  Real protection_mu = 1e-12;
+  if (_harmonic_interpolation)
+  {
+    const auto inv_mu_neighbor = 1.0 / (mu_neighbor + protection_mu);
+    mu_face = 1.0 / (coef_elem * (1/(mu_elem + protection_mu) - inv_mu_neighbor) + inv_mu_neighbor);
+  }
+  else
+    mu_face = coef_elem * (mu_elem - mu_neighbor) + mu_neighbor;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C:161]: You can actually use an interpolation parameter and collapse this if statement into one line. Take a look at `FVDiffusion`.  I tested and `coef_elem * (mu_elem - mu_neighbor) + mu_neighbor` is not faster than the one used in the interpolation function. I suspect the compiler takes care of that in the background.

File: modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+  ADRealTensorValue gradient;
+  if (_dim == 1)
+  {
+    //const auto & grad_u = _u_var->adGradSln(face_info, skewness_correction);
+    const auto & grad_u = _u_var->gradient(face);
+    gradient = ADRealTensorValue(grad_u);
+  }
+  else if (_dim == 2)
+  {
+    // const auto & grad_u = _u_var->adGradSln(face_info, skewness_correction);
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C:111]: Remove unused code

File: modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C [hunk 19]
```diff
@@ -0,0 +1,236 @@
...
+  _u_var(dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("u", 0))),
+  _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("v", 0))
+               : nullptr),
+  _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("w", 0))
+               : nullptr),
+  _harmonic_interpolation(getParam<bool>("harmonic_interpolation"))
+{
+  if ((_var.faceInterpolationMethod() == Moose::FV::InterpMethod::SkewCorrectedAverage) &&
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C:45]: See comment in the `INSFV` version

File: modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C [hunk 20]
```diff
@@ -0,0 +1,236 @@
...
+    velocity(2) = (*_w_var)(elem_face);
+
+  /// ---------------------------------------------------------
+
+  /// ------ Interpolating the diffusion coefficient with different methods ------
+
+  auto mu_elem = _mu(elem_face);
+  auto mu_neighbor = _mu(neighbor_face);
+
+  const Real distance_to_neighbor = _face_info->dCN().norm();
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C:145]: Simplify this using the suggestions in the `INSFV` version

File: modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C [hunk 21]
```diff
@@ -0,0 +1,236 @@
...
+  const Real distance_to_element   = _face_info->eCN().norm();
+  const Real total_distance        = distance_to_neighbor + distance_to_element;
+
+  // Ideally, we would do the following, but we optimize some operations:
+    //   const Real coef_elem = distance_to_neighbor / total_distance;
+    //   const Real coef_neighbor = distance_to_element / total_distance;
+
+    //   ADReal mu_face;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C:153]: Remove unused code

File: modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C [hunk 22]
```diff
@@ -0,0 +1,236 @@
...
+  }
+  else
+    mu_face = coef_elem * (mu_elem - mu_neighbor) + mu_neighbor;
+
+  /// ---------------------------------------------------------
+      
+  /// ---------- Laplacian and Non Laplacian part of diffusion -----------
+
+  // Laplacian part: \frac{\partial u_i}{\partial x_j} n_j
+  const ADReal laplacian_part = _var.gradient(face) * _face_info->normal();
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusionComplete.C:175]: Use the same normal as in the non_laplacian part (just esthetics)

File: modules/navier_stokes/src/fvkernels/PINSFVTKEDSourceSink.C [hunk 23]
```diff
@@ -0,0 +1,194 @@
...
+      auto Sij_12 = 0.5 * (grad_v(2) + grad_w(1));
+      auto Sij_21 = Sij_12;
+      auto Sij_22 = 0.5 * (grad_w(2) + grad_w(2));
+      symmetric_strain_tensor_norm += (std::pow(Sij_02, 2) + std::pow(Sij_20, 2) 
+                                      + std::pow(Sij_12, 2) + std::pow(Sij_21, 2) 
+                                      + std::pow(Sij_22, 2));
+    }
+  }
+
+  //_console << "Linearized ? : " << _linearized_model << std::endl;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVTKEDSourceSink.C:121]: Remove this

File: modules/navier_stokes/src/fvkernels/PINSFVTKEDSourceSink.C [hunk 24]
```diff
@@ -0,0 +1,194 @@
...
+    production = _C1_eps(makeElemArg(_current_elem))
+                      * production_k
+                      * _linear_variable(makeElemArg(_current_elem));
+
+    destruction = _C2_eps(makeElemArg(_current_elem))
+                      * _rho(makeElemArg(_current_elem))
+                      * _var(makeElemArg(_current_elem))
+                      * _linear_variable(makeElemArg(_current_elem));
+    
+    // _console << "Epsilon: " << _var(makeElemArg(_current_elem)) << std::endl;
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVTKEDSourceSink.C:141]: Remove unused code

File: modules/navier_stokes/src/fvkernels/PINSFVTKEDSourceSink.C [hunk 25]
```diff
@@ -0,0 +1,194 @@
...
+  //   //auto bottom_constraint = 0.09*std::pow(_k(makeElemArg(_current_elem)), 1.5) / _max_mixing_length / 10.0; // Must include hard value for C_mu
+  //   //auto top_constraint = std::pow(_k(makeElemArg(_current_elem)), 1.5) / _max_mixing_length * 10.0;
+  //   //residual = (residual > top_constraint) ? residual : top_constraint;
+  //   //residual = (residual < bottom_constraint) ? residual : bottom_constraint;
+  //   if ( (residual < 0) && (std::abs(residual) > 0.5*_var(makeElemArg(_current_elem))) )
+  //     residual *= 0.5*_var(makeElemArg(_current_elem))  / residual;
+  //   Real scaled_argument = 1e-9/std::abs(residual.value() + 1e-12);
+  //   auto exp_argument = std::min(scaled_argument, 20.0);
+  //   residual *= std::exp(- exp_argument);
+  // }
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVTKEDSourceSink.C:188]: I'll stop adding `remove unused code` comments because my browser is getting slower

File: modules/navier_stokes/src/fvkernels/PINSFVTurbulentAdvection.C [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+  return params;
+}
+
+PINSFVTurbulentAdvection::PINSFVTurbulentAdvection(const InputParameters & params)
+  : INSFVAdvectionKernel(params), _rho(getFunctor<ADReal>(NS::density))
+{
+}
+
+ADReal
+PINSFVTurbulentAdvection::computeQpResidual()
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVTurbulentAdvection.C:33]: Why do we need a new kernel for turbulent advection? Is there a specific need behind this? Do we need a rho to make sure we are advecting conserved variables? I think we should extend `INSFVScalarFieldAdvection` instead of creating a 80% redundant kernel.

File: modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C [hunk 27]
```diff
@@ -0,0 +1,80 @@
...
+
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
+}
+
+PINSFVTurbulentDiffusion::PINSFVTurbulentDiffusion(const InputParameters & params)
+  : FVFluxKernel(params),
+    _mu_t(getFunctor<ADReal>("mu_t")),
+    _eps(getFunctor<ADReal>(NS::porosity)),
+    _turb_coef(getFunctor<ADReal>("turb_coef")),
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C:40]: What is this coefficient? Something computed from the turbulent schmidt/prandtl numbers?

File: modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C [hunk 28]
```diff
@@ -0,0 +1,80 @@
...
+      "effective_diffusivity",
+      false,
+      "Whether the diffusivity should be multiplied by porosity, or whether the provided "
+      "diffusivity is an effective diffusivity taking porosity effects into account");
+
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
+}
+
+PINSFVTurbulentDiffusion::PINSFVTurbulentDiffusion(const InputParameters & params)
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C:36]: The comment is the same as before, would be cool to do turbulent and non-turbulent kernels together. In this case the only additional functionalities are the `turb_coeff` and the porosity switch, right?

File: modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C [hunk 29]
```diff
@@ -0,0 +1,80 @@
...
+ADReal
+PINSFVTurbulentDiffusion::computeQpResidual()
+{
+  // Interpolate thermal conductivity times porosity on the face
+  ADReal diffusion_face;
+  const auto face_elem = elemFromFace();
+  const auto face_neighbor = neighborFromFace();
+
+  if (!_porosity_factored_in)
+    Moose::FV::interpolate(Moose::FV::InterpMethod::Average,
```
Comment (grmnptr) [inline modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C:62]: You can do harmonic interpolation here as well. I expect `mut/eps` to be highly discontinuous so I suspect having the option for harmonic interpolation is good here.

Comment (grmnptr) [reply modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C:62]: Also this can be collapsed into one line if you define the `elem_value` and `neighbor_value`-s above with the if statement. Then you can do one interpolation.

File: modules/navier_stokes/test/tests/finite_volume/ins/multiapp-scalar-transport/scalar-transport.i [hunk 30]
```diff
@@ -74,11 +74,11 @@ velocity_interp_method='rc'
     coeff = ${diff}
     variable = scalar
   []
-  [scalar_src]
-    type = FVBodyForce
-    variable = scalar
-    value = 0.1
-  []
+  # [scalar_src]
```
Comment (grmnptr) [inline modules/navier_stokes/test/tests/finite_volume/ins/multiapp-scalar-transport/scalar-transport.i:77]: Why this change?

File: framework/include/dampers/BoundingValueElementDamperFV.h [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Moose Includes
+#include "ElementDamperFV.h"
```
Comment (grmnptr) [inline framework/include/dampers/BoundingValueElementDamperFV.h:13]: Is suggest creating a separate PR for the Dampers. I can help if needed.

**Review by grmnptr — CHANGES_REQUESTED** [review]
I gave it an initial look. Nice job Mauricio!
A few major comments:
- Make sure to include proper docstrings (some of them are compy-pasted at the moment)
- Make sure you add tests to the newly added versions

I added a few minor comments below, mostly about deleting unused code plus formatting stuff. 
I'll come back and give you an in-depth review once these comments are addressed and the PR starts passing tests. Again, great job!

File: framework/include/variables/MooseVariableFV.h [hunk 32]
```diff
@@ -457,12 +462,15 @@ class MooseVariableFV : public MooseVariableField<OutputType>
 
   void setActiveTags(const std::set<TagID> & vtags) override;
 
+  // Increment function for damping
+  const FieldVariableValue & increment() const { return _element_data->increment(); }
+
+protected:
```
Comment (cticenhour) [inline framework/include/variables/MooseVariableFV.h:468]: Does `isInternalFace` really need to be a protected method? This branch is causing a build failure on my local machine, as it is used in `FVAnisotropicDiffusion.C:61`.

**Comment by lindsayad** [comment]
@tanoret do you think you'll push your most recent work? I think you've done a lot of work in the last two months on this, right?
